internal/query/tenants.go

```diff
@@ -161,126 +152,30 @@ func builtInRoles(res *Resource) []*pb.RelationshipUpdate {
 	return rels
 }
 
-type Resource struct {
-	URN          string
-	ID           string
-	ResourceType *ResourceType
-	Fields       map[string]string
-}
-
-type ResourceType struct {
-	Name            string
-	URNNamespace    string
-	URNResourceType string
-	DBType          string
-	Relationships   []*ResourceRelationship
-}
-
-type ResourceRelationship struct {
-	Name       string
-	Field      string
-	Optional   bool
-	DBTypes    string
-	DBRelation string
-}
-
-func GetResourceTypes() []*ResourceType {
-	return []*ResourceType{
+func GetResourceTypes() []types.ResourceType {
+	return []types.ResourceType{
 		{
-			Name:            "Tenant",
-			DBType:          "tenant",
-			URNNamespace:    "infratographer",
-			URNResourceType: "tenant",
-			Relationships: []*ResourceRelationship{
+			Name: "tenant",
+			Relationships: []types.ResourceRelationship{
 				{
-					Name:       "Parent tenant",
-					Field:      "parent_tenant_id",
-					DBTypes:    "tenant",
-					DBRelation: "parent",
-					Optional:   true,
+					Name:     "tenant",
+					Type:     "tenant",
+					Optional: true,
```

Comment:
Given that the subject resource is being deleted from here, do you have a notion of how we'll register subjects? Would we auto-create them as role assignments happen?

Reply:
This got me thinking - RESTful management of resources may not necessarily be what we want here. Resources in SpiceDB are defined only in terms of their relationships, so it's not possible to create a resource that doesn't have relationships.

It may be better to rework this so that the resource API doesn't manage resources, but rather manages roles and relationships.